mmocr/datasets/preparers/config_generator.py

```diff
@@ -317,6 +317,108 @@ def _gen_dataset_config(self) -> str:
         return cfg
 
 
+@CFG_GENERATORS.register_module()
+class TextRecogLMDBConfigGenerator(TextRecogConfigGenerator):
```

Comment:
The current design introduces a new config generator just for a new file format. It's easy to use, but not developer-friendly enough, since it could be hard to tell the border where config generators are divided.

For simplicity, we want to keep only one config generator for each task. A straightforward way is to deduce the file format from the filename passed to `train_ann`, `val_ann` and `test_ann`, then generate the config for each split accordingly.